examples/mesh_calibration_api.py

```diff
@@ -0,0 +1,267 @@
+        cam_right = pipeline.createMonoCamera()
+    else:
+        cam_left = pipeline.createXLinkIn()
+        cam_right = pipeline.createXLinkIn()
+    stereo = pipeline.createStereoDepth()
+    xout_left = pipeline.createXLinkOut()
+    xout_right = pipeline.createXLinkOut()
+    xout_depth = pipeline.createXLinkOut()
+    xout_rectif_left = pipeline.createXLinkOut()
+    xout_rectif_right = pipeline.createXLinkOut()
+
+    if from_camera:
+        cam_left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+        cam_right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+        res = (
+            dai.MonoCameraProperties.SensorResolution.THE_720_P
+            if resolution[1] == 720
+            else dai.MonoCameraProperties.SensorResolution.THE_400_P
+        )
+        for mono_cam in (cam_left, cam_right):  # Common config
+            mono_cam.setResolution(res)
+            # mono_cam.setFps(20.0)
+    else:
+        cam_left.setStreamName("in_left")
+        cam_right.setStreamName("in_right")
+
+    stereo.setOutputDepth(out_depth)
+    stereo.setOutputRectified(out_rectified)
+    stereo.setConfidenceThreshold(200)
+    stereo.setRectifyEdgeFillColor(0)  # Black, to better see the cutout
+    stereo.setMedianFilter(median)  # KERNEL_7x7 default
+    stereo.setLeftRightCheck(lrcheck)
+    stereo.setExtendedDisparity(extended)
+    stereo.setSubpixel(subpixel)
+    if from_camera:
+        # Default: EEPROM calib is used, and resolution taken from MonoCamera nodes
+        # stereo.loadCalibrationFile(path)
+        pass
+    else:
+        stereo.setEmptyCalibration()  # Set if the input frames are already rectified
+        stereo.setInputResolution(1280, 720)
+
+    xout_left.setStreamName("left")
+    xout_right.setStreamName("right")
+    xout_depth.setStreamName("depth")
+    xout_rectif_left.setStreamName("rectified_left")
+    xout_rectif_right.setStreamName("rectified_right")
+
+    cam_left.out.link(stereo.left)
+    cam_right.out.link(stereo.right)
+    stereo.syncedLeft.link(xout_left.input)
+    stereo.syncedRight.link(xout_right.input)
+    stereo.depth.link(xout_depth.input)
+    if out_rectified:
+        stereo.rectifiedLeft.link(xout_rectif_left.input)
+        stereo.rectifiedRight.link(xout_rectif_right.input)
+
+    streams = ["depth"]
+    if out_rectified:
+        streams.extend(["rectified_left", "rectified_right"])
+    streams.extend(["left", "right"])
+
+    parent_path = Path(__file__).resolve().parents[1]
+    mesh_directory = str(parent_path) + "/generated" if mesh_directory is None else mesh_directory
+    generate_mesh(calibData, mesh_directory)
+
+    # Load mesh files
+    left_mesh = mesh_directory + "/left_mesh.calib"
+    right_mesh = mesh_directory + "/right_mesh.calib"
+    stereo.loadMeshFiles(left_mesh, right_mesh)
+
+    return pipeline, streams
+
+
+def generate_mesh(calibData, output_path):
+    M1 = np.array(calibData.getCameraIntrinsics(dai.CameraBoardSocket.LEFT))
```

Comment:
intrinsics changes based on resolution. So pass the resolution too when fetching them.